### PR TITLE
Add new `area` and `need-X` labels

### PR DESCRIPTION
**Pull request by a-hilaly**
Issue #, if available:

Description of changes:

Add these new labels:
- `area/helm`: Issues or PRs related to Helm charts
- `area/testing`: Issues or PRs related to testing
- `area/resource-references`: Issues or PRs related to resource references
- `need-investigation`: Indicates an issue needs some investigation.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prow/jobs/jinja/config/label_sync.jinja2 [hunk 1]
```diff
@@ -322,6 +337,11 @@ default:
       target: prs
       prowPlugin: require-matching-label
       addedBy: prow
+    - color: e11d21
+      description: Indicates an issue needs some investigation.
+      name: needs-investigation
```
Comment (jljaco) [inline prow/jobs/jinja/config/label_sync.jinja2:342]: `priority/needs-investigation`?

File: prow/jobs/jinja/config/label_sync.jinja2 [hunk 2]
```diff
@@ -114,6 +124,11 @@ default:
       name: area/security
       target: both
       addedBy: label
+    - color: 0052cc
+      description: Issues or PRs related to testing
+      name: area/testing
```
Comment (jljaco) [inline prow/jobs/jinja/config/label_sync.jinja2:129]: seems pretty close to `area/infra` in actual usage.

**Review by jljaco — APPROVED** [review]
Approved with some minor nits

File: prow/jobs/jinja/config/label_sync.jinja2 (hunk 2 above)
Comment (a-hilaly) [reply prow/jobs/jinja/config/label_sync.jinja2:129]: We have few issues that are related to testing without necessarily being related to prow or infrastructure. `area/infra` is a very generalistic one that don't necessarily include testing. I personally think label will be useful.

File: prow/jobs/jinja/config/label_sync.jinja2 (hunk 1 above)
Comment (a-hilaly) [reply prow/jobs/jinja/config/label_sync.jinja2:342]: We have the same `priority/*` labels as the ones defined in kubernetes. I think we should keep it that way + we already have `priority/awaiting-more-evidence` which doesn't necessarily mean that we *need* to investigate an issue. Thoughts?

**Pull request merged**